File: address-geocode-us-python-examples/get_best_match_v4_soap_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../address-geocode-us-python/SOAP"))

from get_best_match_v4_soap import GetBestMatchV4Soap

def get_best_match_v4_soap_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n----------------------------------------------")
    print("Address Geocode US - GetBestMatchV4 - SOAP SDK")
    print("----------------------------------------------")

    address = "136 W Canon Perdido St Ste D"
    city = "Santa Barbara"
    state = "CA"
    postal_code = "93101"
    timeout_seconds = 15

    print("\r\n* Input *\r\n")
    print(f"Address        : {address}")
    print(f"City           : {city}")
    print(f"State          : {state}")
    print(f"PostalCode     : {postal_code}")
    print(f"License Key    : {license_key}")
    print(f"Is Live        : {is_live}")
    print(f"Timeout Seconds: {timeout_seconds}")

    try:
        service = GetBestMatchV4Soap(address, city, state, postal_code, license_key, is_live, timeout_seconds)
        response = service.get_best_match_v4()

        if not hasattr(response, 'Error') or not response.Error:
            print("\r\n* Geocode Info *\r\n")
            print(f"Level            : {response.Level}")
            print(f"Level Description: {response.LevelDescription}")
            print(f"Latitude         : {response.Latitude}")
            print(f"Longitude        : {response.Longitude}")
            print(f"Zip              : {response.Zip}")

            print("\r\n* Information Components *\r\n")
            if hasattr(response, 'InformationComponents') and response.InformationComponents:
                for component in response.InformationComponents.InformationComponent:
                    print(f"{component.Name}: {component.Value}")
            else:
                print("No information components found.")
        else:
            print("No geocode info found.")

        if hasattr(response, 'Error') and response.Error:
            print("\r\n* Error *\r\n")
            print(f"Error Desc    : {response.Error.Desc}")
            print(f"Error Number  : {response.Error.Number}")
            print(f"Error Location: {response.Error.Desc}")

    except Exception as e:
        print("\r\n* Error *\r\n")
        print(f"Exception occurred: {str(e)}")

File: address-geocode-us-nodejs-examples/get_best_match_v4_soap_sdk_example.js
import { GetBestMatchV4Soap } from "../address-geocode-us-nodejs/SOAP/get_best_match_v4_soap.js";

async function GetBestMatchV4SoapGo(licenseKey, isLive) {
    console.log("\n----------------------------------------------");
    console.log("Address Geocode US - GetBestMatchV4 - SOAP SDK");
    console.log("----------------------------------------------");

    const address = "136 W Canon Perdido St Ste D";
    const city = "Santa Barbara";
    const state = "CA";
    const postalCode = "93101";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Address        : ${address}`);
    console.log(`City           : ${city}`);
    console.log(`State          : ${state}`);
    console.log(`PostalCode     : ${postalCode}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try
    {
        const agus = new GetBestMatchV4Soap(address, city, state, postalCode, licenseKey, isLive, timeoutSeconds);
        const response = await agus.invokeAsync();
    
        console.log("\n* Geocode Info *\n");
        if (response)
        {
            console.log(`Level            : ${response.Level}`);
            console.log(`Level Description: ${response.LevelDescription}`);
            console.log(`Latitude         : ${response.Latitude}`);
            console.log(`Longitude        : ${response.Longitude}`);
            console.log(`Zip              : ${response.Zip}`);

            console.log("\n* Information Components *\n");
            if (response?.InformationComponents.InformationComponent.length > 0)
            {
                const raw = response?.InformationComponents?.InformationComponent;
                const components = Array.from(raw ?? []);
                components.forEach((component, index) => {
                    console.log(`${component.Name}: ${component.Value}`);
                });
            }
            else
            {
                console.log("No information components found.");
            }
        }
        else
        {
            console.log("No geocode info found.");
        }

        if (response.Error)
        {
            console.log("\n* Error *\n");
            console.log(`Error Description: ${response.Error.Desc}`);
            console.log(`Error Number     : ${response.Error.Number}`);
            console.log(`Error Location   : ${response.Error.Location}`);
        }
    }
    catch (e)
    {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export {GetBestMatchV4SoapGo };
